src/program.fs

```diff
@@ -88,19 +104,38 @@ module Program =
         { program
             with onError = onError }
 
+#if FABLE
+    type private CachedValue<'a>() =
+        [<DefaultValue>] val mutable Value:'a
+        member this.SetValue(generator: unit ->'a) =
+            if weakEqual this.Value null then
```

Comment:
That line might need @alfonsogarciacaro I expected `[<DefaultValue>]` to get me the same value as `Unchecked.defaultof<'a>` (So `null`) but it's not the case currently. The field isn't set at alls and so it's `undefined`.

That's why I used JS weak equality because it sounds like a bug (or at least a quirk) that might be one day changed, and JS equality cover both.

Reply:
Hmm, I think we're not covering `[<DefaultValue>]`. Could you please send a failing test to Fable? About weak equality, when using the `=` operator with `null/None` Fable uses weak equality in JS as it doesn't make a distinction between `undefined` or `null` (for example, in the case of TS optional members).

Reply:
> Fable uses weak equality in JS as it doesn't make a distinction between undefined or null (for example, in the case of TS optional members).

Ah cool, I can't use it here as `= null` doesn't work between lambdas in F# (They are neither nullable nor equatable) but it's good to know that in a normal case Fable would have done the right thing.

> Could you please send a failing test to Fable?

I'll do a PR. It's essentially:

```f#
type Weird() =
    [<DefaultValue>] val mutable Value:int

let ok = Weird().Value = 0
```